camayoc/tests/qpc/ui/utils.py

```diff
@@ -199,6 +216,63 @@ def delete_credential(view, names):
                 xpath=row_xpath(name)), timeout=1)
 
 
+def edit_credential(view, original_name, options):
+    """Edit a credential through the UI and verify it was edited.
+
+    :param view: The view context (should be the browser view)
+    :param original_name: The original name of the credential.
+    :param options: The options to be edited within the credential.
+    """
+    view.refresh()
+    dash = DashboardView(view)
+    dash.nav.select('Credentials')
+    view.wait_for_element(locator=Locator(
+        xpath=(edit_xpath(original_name))))
+    GenericLocatorWidget(view, locator=Locator(
+        xpath=edit_xpath(original_name))).click()
+    modal = CredentialModalView(view, locator=Locator(css='.modal-content'))
+    if 'name' in options:
+        fill(modal, field_xpath('Credential Name'), options['name'])
+    if 'username' in options:
+        fill(modal, field_xpath('Username'), options['username'])
+    if 'sshkeyfile' in options:
+        try:
+            auth_type = Dropdown(view, 'Username and Password')
+            auth_type.item_select('SSH Key')
+        except NoSuchElementException:
+            auth_type = Dropdown(view, 'SSH Key')
+            auth_type.item_select('SSH Key')
+        fill(modal, field_xpath('SSH Key File'), options['sshkeyfile'])
+    elif 'password' in options:
+        fill(modal, field_xpath('Password'), options['password'])
+    if 'become_user' in options:
+        fill(modal, field_xpath('Become User'), options['become_user'])
+    # Hack to deal with the fact that the GET refresh isn't
+    # implemented when the save button is clicked.
+    # https://github.com/quipucords/quipucords/issues/1399
+    # https://github.com/quipucords/camayoc/issues/280
+    wait_for_animation()
+    modal.save_button.click()
+    wait_for_animation()
+    view.refresh()
+    dash.nav.select('Credentials')
+    # Assert the row with the credential name exists.
+    view.wait_for_element(locator=Locator(
+        xpath=row_xpath(options['name'])), delay=0.3, timeout=10)
+    GenericLocatorWidget(view, locator=Locator(
+        xpath=edit_xpath(options['name']))).click()
+    modal = CredentialModalView(view, locator=Locator(css='.modal-content'))
+    # Assert that the changed variables were in fact changed.
+    if 'name' in options:
```

Comment:
If you passed the keys in options as the fields' label then you could use a for loop here.